cmd/clusterctl/pkg/client/cluster/mover.go

```diff
@@ -229,6 +236,72 @@ func setClusterPause(proxy Proxy, clusters []*node, value bool, log logr.Logger)
 	return nil
 }
 
+// ensureNamespaces ensures all the expected target namespaces are in place before creating objects.
+func (o *objectMover) ensureNamespaces(graph *objectGraph, toProxy Proxy) error {
+	cs, err := toProxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	namespaces := sets.NewString()
+	for _, node := range graph.getNodesWithClusterTenants() {
+		namespace := node.identity.Namespace
+
+		// If the namespace was already processed, skip it.
+		if namespaces.Has(namespace) {
+			continue
+		}
+		namespaces.Insert(namespace)
+
+		// Otherwise check if namespace exists (also dealing with RBAC restrictions).
+		ns := &corev1.Namespace{}
+		key := client.ObjectKey{
+			Name: namespace,
+		}
+
+		if err := cs.Get(ctx, key, ns); err == nil {
+			return nil
+		}
+		if apierrors.IsForbidden(err) {
+			namespaces := &corev1.NamespaceList{}
+			if err := cs.List(ctx, namespaces); err != nil {
+				return err
+			}
+
+			namespaceExists := false
+			for _, ns := range namespaces.Items {
+				if ns.Name == namespace {
+					namespaceExists = true
+					break
+				}
+			}
+			if namespaceExists {
+				continue
+			}
```

Comment:
What I meant was to check `NamespaceList` Continue flag and use `client.Continue` in the List call

Reply:
https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machine_controller_noderef.go#L95-L120

Reply:
ok, now I got it